diff --git a/wsutil/privileges.h b/wsutil/privileges.h
index 8ed8e5480a..6ceed0a6cb 100644
--- a/wsutil/privileges.h
+++ b/wsutil/privileges.h
@@ -23,6 +23,26 @@
 #ifndef __PRIVILEGES_H__
 #define __PRIVILEGES_H__
 
+/* Fetch the real user ID, effective user ID, and saved-set user ID,
+   of the calling process.  */
+extern int getresuid (__uid_t *__ruid, __uid_t *__euid, __uid_t *__suid)
+     __THROW;
+
+/* Fetch the real group ID, effective group ID, and saved-set group ID,
+   of the calling process.  */
+extern int getresgid (__gid_t *__rgid, __gid_t *__egid, __gid_t *__sgid)
+     __THROW;
+
+/* Set the real user ID, effective user ID, and saved-set user ID,
+   of the calling process to RUID, EUID, and SUID, respectively.  */
+extern int setresuid (__uid_t __ruid, __uid_t __euid, __uid_t __suid)
+     __THROW __wur;
+
+/* Set the real group ID, effective group ID, and saved-set group ID,
+   of the calling process to RGID, EGID, and SGID, respectively.  */
+extern int setresgid (__gid_t __rgid, __gid_t __egid, __gid_t __sgid)
+     __THROW __wur;
+
 #include "ws_symbol_export.h"
 
 #ifdef __cplusplus
