diff --git a/epan/ftypes/ftype-time.c b/epan/ftypes/ftype-time.c
index aea0bd6f77..e9891a3744 100644
--- a/epan/ftypes/ftype-time.c
+++ b/epan/ftypes/ftype-time.c
@@ -24,6 +24,10 @@
 #include <stdlib.h>
 #include <string.h>
 
+extern char *strptime_l (const char *__restrict __s,
+                         const char *__restrict __fmt, struct tm *__tp,
+                         __locale_t __loc) __THROW;
+
 /*
  * Just make sure we include the prototype for strptime as well
  * (needed for glibc 2.2) but make sure we do this only if not
